src/clarity-angular/forms/checkbox/checkbox.ts

```diff
@@ -55,39 +68,51 @@ export class Checkbox implements ControlValueAccessor {
     // Support for inline checkboxes, adds the necessary class to the host
     @Input("clrInline") public inline = false;
 
-    @Input("clrChecked")
-    _checked = false;
+    private _checked = false;
 
     public get checked() {
         return this._checked;
     }
 
+    @Input("clrChecked")
     public set checked(value: boolean) {
         if (value !== this._checked) {
-            this.indeterminate = false;
-            this._checked = value;
-            this.change.emit(this.checked);
+            this.setChecked(value);
+            if (this._indeterminate) {
+                this.setIndeterminate(false);
+            }
```

Comment:
You're changing the order of the event emitters here, I remember vaguely the order was finely tuned because it created issues otherwise. If all the unit tests are working it should be fine, but I'm just pointing out there is a risk here.